langchain/chains/qa_with_sources/vector_db.py

```diff
@@ -15,11 +17,33 @@ class VectorDBQAWithSourcesChain(BaseQAWithSourcesChain, BaseModel):
     """Vector Database to connect to."""
     k: int = 4
     """Number of results to return from store"""
+    reduce_k_below_max_tokens: bool = False
+    """Reduce the number of results to return from store based on tokens limit"""
+    max_tokens_limit: int = 3375
+    """Restrict the docs to return from store based on tokens,
+    enforced only for StuffDocumentChain and if reduce_k_below_max_tokens is to true"""
     search_kwargs: Dict[str, Any] = Field(default_factory=dict)
     """Extra search args."""
 
+    def _page_content(self, doc: Document) -> str:
+        return doc.page_content
+
+    def _reduce_tokens_below_limit(self, docs: List[Document]) -> List[Document]:
+        if isinstance(self.combine_documents_chain, StuffDocumentsChain):
```

Comment:
can you also add a check here to see if the `reduce_k_below_max_tokens` flag is set?